helm-chart/templates/01-service-account.yaml

```diff
@@ -9,4 +9,4 @@ metadata:
     kubeshark-created-by: kubeshark
     kubeshark-managed-by: kubeshark
   name: kubeshark-service-account
-  namespace: '{{ .Values.tap.selfnamespace }}'
+  namespace: '{{ default "kubeshark" .Release.Namespace }}'
```

Comment:
The template files are generated automatically. So, please don't modify them directly. See the first line:

https://github.com/kubeshark/kubeshark/blob/08751add1180e7d89bada9228b059739d1e70b27/helm-chart/templates/01-service-account.yaml#L1

The correct way to modify the line is through changing this line:

https://github.com/kubeshark/kubeshark/blob/2563cc19221fb5007411613694777c0e80b717c1/cmd/helmChart.go#L116

Related to https://github.com/kubeshark/kubeshark/issues/1349

We have an existing way of configuring the Kubeshark deployment through `config.yaml` / `kubeshark.yaml` file. So, the namespace should be sourced from the `config.yaml`. [See the docs](https://docs.kubeshark.co/en/config).

Reply:
To clarify; we should generate such a template directive that prioritizes `.Release.Namespace` over `.Values.tap.selfnamespace`. We should also not remove `00-namespace.yaml` but instead generate such a template directive that disables `00-namespace.yaml` if `.Release.Namespace` is present or `.Values.tap.selfnamespace` is something other than `kubeshark`.